### PR TITLE
 Fixes #24138 - Trigger tasks in batches

### PR DESCRIPTION
**Pull request by adamruzicka**
Requires:
- [x] https://github.com/theforeman/smart_proxy_dynflow/pull/55
- [x] https://github.com/theforeman/foreman-tasks/pull/369

### CODE REVIEW

**Comment by bkearney** [comment]
@adamruzicka what is the status of this?

**Comment by adamruzicka** [comment]
This is waiting for the foreman-tasks dependency https://github.com/theforeman/foreman-tasks/pull/385

**Comment by iNecas** [comment]
I've released new foreman-tasks/core and bumped dependency here: let's see how the tests go

File: lib/foreman_remote_execution_core/polling_script_runner.rb [hunk 1]
```diff
@@ -3,8 +3,8 @@ class PollingScriptRunner < ScriptRunner
 
     DEFAULT_REFRESH_INTERVAL = 60
 
-    def initialize(options, user_method)
-      super(options, user_method)
+    def initialize(options, user_method, suspended_action: nil)
```
Comment (iNecas) [inline lib/foreman_remote_execution_core/polling_script_runner.rb]: @adamruzicka I needed to add this to fix the async mode

**Comment by iNecas** [comment]
During testing, I've noticed that disabling of the batch mode actually was not working, so fixed it here https://github.com/theforeman/foreman-tasks/pull/397/files (+ reduced the batch size to a bit smaller number).

All my tests worked well at the end: I was a bit surprised that turning the `async` mode on proxy influenced the way how ansible is reporting (I would expect that affecting ssh only), but that's not related to this PR.

**Comment by adamruzicka** [comment]
Looks good to me

**Comment by iNecas** [comment]
Changed squashed: merging

**Comment by iNecas** [comment]
Thanks @adamruzicka

**Pull request merged**